rotkehlchen/chain/ethereum/modules/superchain_bridge/decoder.py

```diff
@@ -115,16 +130,37 @@ def _decode_prove_withdrawal(self, context: DecoderContext) -> DecodingOutput:
             balance=Balance(),
             location_label=context.transaction.from_address,
             notes=f'Prove optimism bridge withdrawal 0x{withdrawal_hash}',
-            counterparty=CPT_OPTIMISM,
+            counterparty=self.get_counterparty_details(target_chain).identifier,
             address=context.tx_log.address,
         )
         return DecodingOutput(event=event)
 
+    @staticmethod
+    def get_target_chain(address: ChecksumEvmAddress) -> ChainID:
+        """Get the target chain for a bridge address."""
+
+        for addresses, chain_id in BRIDGE_ADDRESSES_TO_TARGET_CHAINS.items():
+            if address in addresses:
+                return chain_id
+
+        raise ValueError(f'Unsupported bridge address: {address}')
+
+    @staticmethod
+    def get_counterparty_details(chain: ChainID) -> CounterpartyDetails:
+        """Get the counterparty for a bridge chain."""
+
+        if chain == ChainID.OPTIMISM:
+            return OPTIMISM_CPT_DETAILS
+        if chain == ChainID.BASE:
+            return BASE_CPT_DETAILS
+
+        raise ValueError(f'Unsupported chain id: {chain}')
```

Comment:
We never raise ValueErrror in our codebase. Ever. You probably wanted an assertion in this case. But still this should never happen as the type should have been `Literal[ChainID.OPTIMISM, ChainID.BASE]`